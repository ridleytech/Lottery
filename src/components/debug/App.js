/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow strict-local
 */

import React, {Component} from 'react';
import {
  SafeAreaView,
  StyleSheet,
  ScrollView,
  StatusBar,
  FlatList,
  Text,
} from 'react-native';

import Home from './src/Home';
import data1 from './src/data';
import PeopleItem from './src/PeopleItem';
import Move from './src/Move';

//console.log('data: ' + JSON.stringify(data1.items));

data1.items.map((ob) => {
  console.log('ob: ' + ob.name);
});

class App extends Component {
  dividNum = (x, y) => x / y;

  checkZero = (func) => (...args) => {
    if (args[1] === 0) {
      return null;
    } else {
      return func(...args);
    }
  };

  checkEven = (x) => {
    return x % 2 === 0;
  };

  ascending = (a, b) => {
    if (a > b) {
      return 1;
    } else if (a < b) {
      return -1;
    } else {
      return 0;
    }
  };

  addNumbers = (acc, cv) => {
    return acc + cv;
  };

  loop = () => {
    for (var i = 0; i < 10; i++) {
      console.log('i: ' + i);
    }
  };

  map = (arr, func) => arr.reduce((acc, cv) => [...acc, func(cv)], []);

  arr = [6, 2, 5, 1, 3, 4];

  componentDidMount() {
    //this.loop();
    // const test = this.checkZero(this.dividNum);
    // console.log(test(43, 2));
    // const even = this.arr.filter(this.checkEven);
    // console.log(even);
    // const every = this.arr.some(this.checkEven);
    // console.log(every);
    //const sorted = this.arr.slice().sort(this.ascending);
    // console.log(sorted);
    // const reduced = this.arr.reduce(this.addNumbers);
    // console.log('reduced: ' + reduced);
    //console.log('map: ' + this.map([1, 2, 3], (x) => x * 2));
  }

  render() {
    return (
      <>
        <StatusBar barStyle="dark-content" />
        <SafeAreaView>
          {/* <Home /> */}
          <FlatList
            data={data1.items}
            keyExtractor={(item, index) => index.toString()}
            renderItem={(item) => <PeopleItem item={item.item} />}
          />
          <Move />
        </SafeAreaView>
      </>
    );
  }
}

const styles = StyleSheet.create({});

export default App;
