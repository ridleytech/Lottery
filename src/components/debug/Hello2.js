import React, {Component} from 'react';
import {Text, View, Button} from 'react-native';

class Hello2 extends Component<Props> {
  constructor(props: Props) {
    super(props);
  }

  testButton = () => {
    return 2;
  };

  render() {
    return (
      <View style={{flex: 1, alignItems: 'center', justifyContent: 'center'}}>
        <Text testID={'username'}>Loading...</Text>
      </View>
    );
  }
}

export default Hello2;
