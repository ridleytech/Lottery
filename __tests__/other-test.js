import {Text} from 'react-native';
import React from 'react';
import App from '../App';
import Hello2 from '../src/components/debug/Hello2';
import Enzyme from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({adapter: new Adapter()});

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import {findNodeHandle} from 'react-native';

import {shallow} from 'enzyme';

describe('a', () => {
  //   it('b', () => {
  //     //const api = renderer.create(<Hello />).root.instance;
  //     const snap = renderer.create(<Hello />).toJSON();
  //     //expect(1 + 2).toBe(3);
  //     expect(snap).toMatchSnapshot();
  //   });

  // it('c', () => {
  //   //const api = renderer.create(<Hello />).root.instance;
  //   const snap = renderer.create(<Hello2 />).getInstance();
  //   //expect(1 + 2).toBe(3);
  //   var val = snap.testButton();

  //   expect(val).toBe(3);
  // });

  findEl = (tree, el) => {
    var result = undefined;

    for (node in tree.children) {
      if (tree.children[node].props.testID === el) {
        result = true;
      }
    }

    return result;
  };

  // it('find', () => {
  //   var tree = renderer.create(<Hello2 />).toJSON();

  //   expect(findEl(tree, 'username')).toBe(true);
  // });

  it('render text', () => {
    var wrapper = shallow(<Text>Hello</Text>);
    expect(wrapper.text()).toBe('Hello1');
  });
});
