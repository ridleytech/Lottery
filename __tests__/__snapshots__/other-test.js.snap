// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a b 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Text>
    Hello
  </Text>
</View>
`;
